columnphysics/icepack_flux.F90

```diff
@@ -184,6 +198,14 @@ subroutine merge_fluxes (aicen,                &
       fsalt     = fsalt     + fsaltn    * aicen
       fhocn     = fhocn     + fhocnn    * aicen
       fswthru   = fswthru   + fswthrun  * aicen
+      if (present(fswthruvdr)) &
+         fswthruvdr   = fswthruvdr   + fswthrunvdr  * aicen
+      if (present(fswthruvdf)) &
+         fswthruvdf   = fswthruvdf   + fswthrunvdf  * aicen
+      if (present(fswthruidr)) &
+         fswthruidr   = fswthruidr   + fswthrunidr  * aicen
+      if (present(fswthruidf)) &
+         fswthruidf   = fswthruidf   + fswthrunidf  * aicen
```

Comment:
If fswthrun* is size 1 and aicen is not, I think this is risky and will fail on some/all compilers.  Recommend adding a check that fswthrun* either matches the size of aicen then this code is run, fswthrun* is size 1 then this code is skipped, and anything else might generate an error.  Thoughts?

Reply:
Fortunately in merge_fluxes, this is called by category and everything is dimension 1. However, I do have a problem in icepack_shortwave.F90 (run_dEdd) here:

            if(present(fswthrun_vdr)) fswthrun_vdr(n) = l_fswthrun_vdr
            if(present(fswthrun_vdf)) fswthrun_vdf(n) = l_fswthrun_vdf
            if(present(fswthrun_idr)) fswthrun_idr(n) = l_fswthrun_idr
            if(present(fswthrun_idf)) fswthrun_idf(n) = l_fswthrun_idf

This is causing a segmentation violation, but it shouldn't be a problem because of the "present"?  We can take this discussion offline.